runtime/src/anchor.rs

```diff
@@ -0,0 +1,142 @@
+//! Minimal proof of existence registry.
+//!
+//! Anchors are hashed once before being added to storage. To check whether an anchor exists
+//! query the "Anchors" map for the hash of the anchor. If a corresponding value exists, then the
+//! anchor exists and the value represents the block number when it was first published.
+
+use alloc::vec::Vec;
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, ensure,
+    traits::Get,
+};
+use frame_system::{self as system, ensure_signed};
+use sp_runtime::traits::Hash;
+
+pub trait Trait: system::Trait {
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+}
+
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+        /// The anchor being posted was already created in a previous block.
+        AnchorExists,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Blob {
+        // Hasher can be the identity here becuse we perform a hash ourself which has the same
+        // merkle-trie balancing effect as using a hash-prefix map.
+        Anchors: map hasher(identity) <T as system::Trait>::Hash =>
```

Comment:
I realized we can limit chain state usage by not storing the entire payload, but just a hash. The user can still submit whatever content they want.

Reply:
> // Hasher can be the identity here becuse we perform a hash ourself which has the same
> // merkle-trie balancing effect as using a hash-prefix map.

@lovesh can you confirm this is true?